eslint.config.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * @author 唯然<weiran.zsd@outlook.com>
+ */
+"use strict"
+
+const js = require("@eslint/js")
+const { FlatCompat } = require("@eslint/eslintrc")
+const globals = require("globals")
+const nodeRecommended = require("eslint-plugin-n/configs/recommended-script")
+
+const compat = new FlatCompat({
+    baseDirectory: __dirname,
+    recommendedConfig: js.configs.recommended,
+})
+
+module.exports = [
+    ...compat.extends(
+        // "eslint:recommended", // TODO: adding this line will cause commonjs globals to be undefined
```

Comment:
This is because of https://github.com/eslint-community/eslint-plugin-n/pull/95/files#r1180506643